### PR TITLE
Dweb 6

### PR DESCRIPTION
**Pull request by vjnathe-webonise**
**What does this PR do?**
Facebook pixel advance matching for external_id from userId or anonymousId if not present under traits, user has to enable option `userIdAsExternalId` defaults to false.

**Are there breaking changes in this PR?**
No.

**Any background context you want to provide?**
Earlier external_id only referenced from traits of identity calls.

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
No.

**Does this require a new integration setting? If so, please explain how the new setting works**
Yes, **userIdAsExternalId** needs to set to **true** which defaults to **false**. Also setting **keyForExternalId** has high preference above this setting.

**Links to helpful docs and other external resources**
https://segment.com/docs/destinations/facebook-pixel/#advanced-matching

### CODE REVIEW

File: integrations/facebook-pixel/lib/index.js [hunk 1]
```diff
@@ -626,6 +626,10 @@ FacebookPixel.prototype.formatTraits = function formatTraits(analytics) {
   if (this.options.keyForExternalId) {
     external_id = traits[this.options.keyForExternalId]; // eslint-disable-line
   }
+  if (!external_id && analytics) { // eslint-disable-line
```
Comment (CarlosMecha) [inline integrations/facebook-pixel/lib/index.js]: Would be possible to enable this using a new option? something like `this.options.userIdAsExternalId`? when we create the documentation for that setting, I'll mention that we fallback to anonymousId if the userId is not present. Facebook Pixel is one of our most used destinations and I would like to give the option to the customers to toggle this new feature :) Thanks!

File: integrations/facebook-pixel/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@segment/analytics.js-integration-facebook-pixel",
   "description": "The Facebook Pixel analytics.js integration.",
-  "version": "2.9.0",
+  "version": "2.9.1",
```
Comment (CarlosMecha) [inline integrations/facebook-pixel/package.json]: Let's do `2.10.0` since we are adding a new feature.

**Review by CarlosMecha — COMMENTED** [review]
Left some comments, good job!

File: integrations/facebook-pixel/test/index.test.js [hunk 3]
```diff
@@ -2748,9 +2751,10 @@ describe('Facebook Pixel', function() {
 
   describe('#formatTraits', function() {
     it('should return an empty object if traits is falsy', function() {
-      var expected = {};
-
       analytics.reset();
+
+      var userData = analytics.user();
+      var expected = { external_id: userData.anonymousId() };
```
Comment (brennan) [inline integrations/facebook-pixel/test/index.test.js]: `analytics.user().anonymousId()` should work here and would be one less line of code. Just a style suggestion. :)

**Comment by vjnathe-webonise** [comment]
@CarlosMecha @brennan Resolved the comments, please have a look.

**Review by CarlosMecha — APPROVED** [review]

**Comment by vjnathe-webonise** [comment]
Closing this PR as PR raised to merge in upstream: https://github.com/segmentio/analytics.js-integrations/pull/296

**Pull request closed without merging**